code/game/objects/items/explosives/bombvest.dm

```diff
@@ -0,0 +1,27 @@
+/obj/item/clothing/suit/storage/marine/harness/boomvest
+	name = "tactical explosive vest"
+	desc = "Obviously someone just strapped a bomb to a marine harness and called it tactical. The light has been removed, and its switch used as the detonator.<br><span class='notice'>Control-Click to set a warcry.</span> <span class='warning'>This harness has no light, toggling it will detonate the vest!</span>"
+	icon_state = "boom_vest"
+	flags_item_map_variant = NONE
+	flags_armor_features = NONE
+	var/bomb_message = ""
+
+//boom
+/obj/item/clothing/suit/storage/marine/harness/boomvest/attack_self(mob/user)
```

Comment:
No parentcall?
Also grabbing them and using them inhand will also explode them

Reply:
its no longer going to explode when it isnt worn.

Reply:
and no parentcall since the parent call would be the light